Program.fs

```diff
@@ -819,42 +963,48 @@ let rec private runner (args : FemtoArgs) =
 
 
 type CLIArguments =
-    | [<MainCommand; ExactlyOnce>] Project of path : string
+    | [<MainCommand>] Project of path : string
     | Validate
     | Resolve
-    | Preview
+    | Version
 
     interface IArgParserTemplate with
 
         member this.Usage =
             match this with
-            | Project _ -> "specify the path to the F# project"
-            | Validate -> "check that the XML tags used in the F# project file are parsable and a npm package version can be calculated"
-            | Resolve -> "resolve and install required packages"
-            | Preview -> "preview required actions for package resolution"
+            | Project _ -> "specify the path to the F# project."
+            | Validate -> "check that the XML tags used in the F# project file are parsable and a npm package version can be calculated."
+            | Resolve -> "resolve and install required packages."
+            | Version -> "display the current version of Femto."
 
 let parseArgs (cliArgs : CLIArguments list) =
     let rec apply (cliArgs : CLIArguments list) (res : FemtoArgs) =
         match cliArgs with
-        | Project project::rest ->
+        | Project project :: rest ->
             let project =
-                if Path.isRelativePath project then
-                    Path.GetFullPath project
+                if IO.Directory.Exists project then
+                    IO.Directory.GetFiles project
+                    |> Seq.tryFind (fun file -> file.EndsWith ".fsproj")
+                    |> Option.map Path.normalizeFullPath
                 else
-                    project
 
-            apply rest { res with Project = Some project }
+                    if Path.isRelativePath project then
+                        Some (Path.GetFullPath project)
+                    else
+                        Some project
+
+            apply rest { res with Project = project }
 
-        | Validate::rest ->
-            apply rest { res with PreviewMetadata = true }
+        | Validate :: rest ->
+            apply rest { res with Validate = true }
 
-        | Resolve::rest ->
+        | Resolve :: rest ->
             apply rest { res with Resolve = true }
 
-        | Preview::rest ->
-            apply rest { res with ResolvePreview = true }
+        | Version :: rest ->
+            apply rest { res with DiplayVersion = true }
 
-        | [ ] ->
+        | _ ->
```

Comment:
I am not a fan of using discard case in DUs. 

The compiler can't help us anymore when adding a new case. Is there another case to support that `[]` here?

Reply:
Yeah you are right, I will add it back